client/blocks/simple-site-rename-form/dialog.jsx

```diff
@@ -0,0 +1,129 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import { noop } from 'lodash';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal Dependencies
+ */
+import Dialog from 'components/dialog';
+import FormLabel from 'components/forms/form-label';
+import FormInputCheckbox from 'components/forms/form-checkbox';
+
+class SiteRenamerConfirmationDialog extends PureComponent {
+	static propTypes = {
+		currentDomainSuffix: PropTypes.string,
+		isVisible: PropTypes.bool,
+		newDomainSuffix: PropTypes.string,
+		onConfirm: PropTypes.func.isRequired,
+		onClose: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		onClose: noop,
+		onConfirm: noop,
+		currentDomainSuffix: '.wordpress.com',
+		newDomainSuffix: '.wordpress.com',
+	};
+
+	state = {
+		isConfirmationChecked: false,
+	};
+
+	toggleConfirmationChecked = () => {
+		this.setState( {
+			isConfirmationChecked: ! this.state.isConfirmationChecked,
+		} );
+	};
+
+	onConfirm = closeDialog => {
+		this.onClose();
+		this.props.onConfirm( this.props.targetSite, closeDialog );
+	};
+
+	onClose = () => {
+		this.props.onClose();
+		this.setState( {
+			isConfirmationChecked: false,
+		} );
+	};
+
+	render() {
+		const {
+			currentDomainName,
+			currentDomainSuffix,
+			isVisible,
+			newDomainName,
+			newDomainSuffix,
+			translate,
+		} = this.props;
+		const buttons = [
+			{
+				action: 'cancel',
+				label: translate( 'Cancel' ),
+			},
+			{
+				action: 'confirm',
+				label: translate( 'Change Site Address' ),
+				onClick: this.onConfirm,
+				disabled: ! this.state.isConfirmationChecked,
+				isPrimary: true,
+			},
+		];
+
+		return (
+			<Dialog
+				className="simple-site-rename-form__dialog"
+				isVisible={ isVisible }
+				buttons={ buttons }
+				onClose={ this.onClose }
+			>
+				<h1>{ translate( "Let's Review" ) }</h1>
+				<p>
+					{ translate(
+						'You are about to change your domain name. Once changed, ' +
```

Comment:
What is the reasoning behind splitting up this string here?

Reply:
I'm 99% sure it's prettier trying to prettify a long string 🤷‍♂️

Reply:
Ah, ok 🤷‍♂️

No idea yet how our translation interfaces work, does this have any implications or will translators just see this as one string? 🤔Unrelated to this PR though, more a general question.

Reply:
I'm fairly sure they'll just see a string as it's all handled by the `i18n` module internally. By the time it's done anything with the string it's been concatted anyhow. (assumptions, purely assumptions)

Reply:
Yes, it will work, see https://github.com/Automattic/i18n-calypso#strings-only. Dynamically constructing the string is a no-no, however (using variables in the string, etc.).